### PR TITLE
feat: added new fields to usage data

### PR DESCRIPTION
**Pull request by ammarkarachi**
*Issue #, if available:*

*Description of changes:*
Added additional fields to the usage data payload.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/context-manager.ts [hunk 1]
```diff
@@ -4,29 +4,50 @@ import { PluginPlatform } from './domain/plugin-platform';
...
+    const frontend = projectConfig['frontend'];
+    projectSettings['frontend'] = frontend;
+    if (projectConfig[frontend] && projectConfig[frontend].framework) {
+      projectSettings['framework'] = projectConfig[frontend].framework;
+    }
+  }
+
+  if (stateManager.localEnvInfoExists()) {
+    const { defaultEditor } = stateManager.getLocalEnvInfo();
+    projectSettings['editor'] = defaultEditor;
```
Comment (cjihrig) [inline packages/amplify-cli/src/context-manager.ts:46]:
```suggestion
    projectSettings.editor = defaultEditor;
```

File: packages/amplify-cli/src/context-manager.ts [hunk 2]
```diff
@@ -4,29 +4,50 @@ import { PluginPlatform } from './domain/plugin-platform';
...
 }
 
 const getVersion = (context: Context) => context.pluginPlatform.plugins.core[0].packageVersion;
 
+const getProjectSettings = (): ProjectSettings => {
+  const projectSettings = {};
+  if (stateManager.projectConfigExists()) {
+    const projectConfig = stateManager.getProjectConfig();
+    const frontend = projectConfig['frontend'];
+    projectSettings['frontend'] = frontend;
```
Comment (cjihrig) [inline packages/amplify-cli/src/context-manager.ts:38]:
```suggestion
    projectSettings.frontend = frontend;
```

File: packages/amplify-cli/src/context-manager.ts [hunk 3]
```diff
@@ -4,29 +4,50 @@ import { PluginPlatform } from './domain/plugin-platform';
...
+    if (projectConfig[frontend] && projectConfig[frontend].framework) {
```
Comment (cjihrig) [inline packages/amplify-cli/src/context-manager.ts:39]: You can simplify this to `projectConfig?.[frontend].framework` and put it in a variable.

File: packages/amplify-cli/src/context-manager.ts [hunk 4]
```diff
@@ -4,29 +4,50 @@ import { PluginPlatform } from './domain/plugin-platform';
...
+      projectSettings['framework'] = projectConfig[frontend].framework;
```
Comment (cjihrig) [inline packages/amplify-cli/src/context-manager.ts:40]:
```suggestion
      projectSettings.framework = projectConfig[frontend].framework;
```

**Review by cjihrig — APPROVED** [review]
LGTM with some optional nits.

File: packages/amplify-cli/src/domain/amplify-usageData/UsageData.ts [hunk 5]
```diff
@@ -3,29 +3,37 @@ import { Input } from '../input';
...
+    this.inputOptions = {};
   }
 
-  init(installationUuid: string, version: string, input: Input): void {
+  init(installationUuid: string, version: string, input: Input, accountId: string, projectSettings: ProjectSettings): void {
     this.installationUuid = installationUuid;
+    this.accountId = accountId;
+    this.projectSettings = projectSettings;
     this.version = version;
+    this.inputOptions = input.options ? _.pick(input.options as InputOptions, ['sandboxId']) : {};
```
Comment (edwardfoyle) [inline packages/amplify-cli/src/domain/amplify-usageData/UsageData.ts:36]: is this only storing the sandboxId? if so, I'd just name it that. Also, does this simplify to:
```suggestion
    this.inputOptions = input?.options?.sandboxId || {};
```

File: packages/amplify-cli/src/domain/amplify-usageData/UsageDataPayload.ts [hunk 6]
```diff
@@ -26,12 +39,26 @@ export class UsageDataPayload {
...
+
+export type ProjectSettings = {
+  frontend?: string;
+  editor?: string;
+  framework?: string;
+};
+
+export type InputOptions = {
+  [key: string]: string | boolean;
+};
```
Comment (edwardfoyle) [inline packages/amplify-cli/src/domain/amplify-usageData/UsageDataPayload.ts:60]:
```suggestion
export type InputOptions = Record<string, string | boolean>;
```

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import AWS from 'aws-sdk';
+import aws from './aws';
+import { loadConfiguration } from '../configuration-manager';
+import { $TSContext } from 'amplify-cli-core';
+
+export class STS {
+  public sts: AWS.STS;
+
+  constructor(private readonly context: $TSContext, options = {}) {
+    const instancePromise = (async () => {
```
Comment (edwardfoyle) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: we need to get away from this pattern of async constructors -- it's really bad practice. Instead follow the getInstance pattern we use for the s3 and amplifyBackend clients

**Review by edwardfoyle — COMMENTED** [review]
left a few nits and some other clarifying questions

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts (hunk 7 above)
Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: Ok

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: So do you want me to change this or not?

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: Ok

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: So do you want me to change this or not?

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: Ok

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: So do you want me to change this or not?

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: So do you want me to change this or not? Your comment is vague here

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: So do you want me to change this or not? Your comment is vague here

Comment (ammarkarachi) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sts.ts:10]: So do you want me to change this or not? Your comment is vague here

File: packages/amplify-cli/src/domain/amplify-usageData/UsageData.ts (hunk 5 above)
Comment (ammarkarachi) [reply packages/amplify-cli/src/domain/amplify-usageData/UsageData.ts:36]: No we are going to extend this in the future, this is there to add more field if needed in the future

**Pull request merged**